test/fuzz/lua/mvcc_test.lua

```diff
@@ -0,0 +1,778 @@
+    local additional_ops = {
+        {
+            type = DQL,
+            subtype = SELECT,
+            key_cnt = 0,
+            fmt = 'box.space.%s.index[%d]:select({}, ' ..
+                  '{iterator = "%s", fullscan = true})',
+        },
+        {
+            type = DQL,
+            subtype = SELECT,
+            key_cnt = 1,
+            fmt = 'box.space.%s.index[%d]:select({%s}, ' ..
+                  '{iterator = "%s", fullscan = true})',
+        },
+    }
+    for _, op in ipairs(additional_ops) do
+        table.insert(tree_idx_meta.ro_ops, op)
+        table.insert(tree_idx_meta.ops, op)
+    end
+    return tree_idx_meta
+end
+
+local function create_hash_idx_meta(fields)
+    local hash_idx_meta = create_generic_idx_meta(fields)
+    hash_idx_meta.iters = {'EQ'}
+    return hash_idx_meta
+end
+
+-- Metadata of tested engines.
+local engines = {
+    {
+        name = 'memtx',
+        space = 'm',
+        idxs = {
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = true}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = true}}),
+            create_hash_idx_meta({{is_nullable = false},
+                                  {is_nullable = false}}),
+        }
+    },
+    {
+        name = 'vinyl',
+        space = 'v',
+        idxs = {
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = true}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = true}}),
+        }
+    },
+}
+
+local function generate_txn_repro_stmt(stmt)
+    return ("tx%d('%s') -- %s\n"):format(stmt.tid, stmt.str,
+                                         json.encode(stmt.res))
+end
+
+local function generate_txn_serial_stmt(stmt)
+    return ("%s -- tx%d: %s\n"):format(stmt.str, stmt.tid,
+                                       json.encode(stmt.res))
+end
+
+local INIT_CODE =
+[[os.execute('rm -rf *.snap *.xlog *.vylog 512 513')
+
+local ffi = require('ffi')
+local fiber = require('fiber')
+local json = require('json')
+local log = require('log')
+local txn_proxy = require('test.box.lua.txn_proxy')
+
+box.cfg{memtx_use_mvcc_engine = %s}
+fiber.set_max_slice({warn = 6000, err = 6000})
+
+]]
+
+local function generate_init_code(memtx_use_mvcc_engine)
+    return (INIT_CODE):format(memtx_use_mvcc_engine and 'true' or 'false')
+end
+
+local SCHEMA_CODE =
```

Comment:
There are no nullable parts without `exclude_null` attribute at all - let's add them.

Reply:
Also, we don't have non-unique indexes here.